src/cpp/session/modules/SessionPythonEnvironments.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * SessionPythonEnvironments.cpp
+ *
+ * Copyright (C) 2020 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include "SessionPythonEnvironments.hpp"
+
+#include <core/Exec.hpp>
+
+#include <session/prefs/UserPrefs.hpp>
+#include <session/prefs/UserPrefValues.hpp>
+
+#include <session/SessionModuleContext.hpp>
+
+using namespace rstudio::core;
+
+namespace rstudio {
+namespace session {
+namespace modules {
+namespace python_environments {
+
+namespace {
+
+} // end anonymous namespace
+
+Error initialize()
+{
+   using namespace module_context;
+   
+   std::string pythonPath = core::system::getenv("RETICULATE_PYTHON");
+   if (pythonPath.empty())
+   {
+      core::system::setenv(
+               "RETICULATE_PYTHON",
+               prefs::userPrefs().pythonDefaultInterpreter());
```

Comment:
If this user pref changes at runtime, do we want to forward that change to `RETICULATE_PYTHON` (if the old value of the var was also derived from the pref)?

Reply:
My exit hatch for this question (resolving whether the environment variable was set by us, versus by the user) was to just require a session restart any time the preference was changed via the UI. (This doesn't help for users who need to change the preference programmatically though, so I think ultimately this is necessary)